### PR TITLE
[MRESOLVER-359] Make build be explicit about build time requirements

### PR DESCRIPTION
**Pull request by cstamas**
Fixes:
* make minimalMavenBuildVersion be "[3.8.7,)", we mean "recent" or maybe "maintained" Maven version.
* make minimalJavaBuildVersion be "[1.8.0-362,)" we mean "recent" Java 8. *CI helped to decide: 362*

Note: values taken from current CI setup.

---

https://issues.apache.org/jira/browse/MRESOLVER-359

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -93,7 +93,8 @@
     <guavaVersion>30.1-jre</guavaVersion>
     <guavafailureaccessVersion>1.0.1</guavafailureaccessVersion>
     <slf4jVersion>1.7.36</slf4jVersion>
-    <minimalMavenBuildVersion>3.8.4</minimalMavenBuildVersion>
+    <minimalMavenBuildVersion>[3.8.4,)</minimalMavenBuildVersion>
+    <minimalJavaBuildVersion>[1.8.0-362,)</minimalJavaBuildVersion>
```
Comment (slawekjaranowski) [inline pom.xml:97]: Here can also be simple `1.8.0-362`

File: pom.xml [hunk 2]
```diff
@@ -93,7 +93,8 @@
     <guavaVersion>30.1-jre</guavaVersion>
     <guavafailureaccessVersion>1.0.1</guavafailureaccessVersion>
     <slf4jVersion>1.7.36</slf4jVersion>
-    <minimalMavenBuildVersion>3.8.4</minimalMavenBuildVersion>
+    <minimalMavenBuildVersion>[3.8.4,)</minimalMavenBuildVersion>
```
[the remark this answers is not in the document]
Comment (cstamas) [reply pom.xml:96]: Is not, see https://maven.apache.org/enforcer/enforcer-rules/versionRanges.html

"3.8.4" was "x >= 3.8.4 but 3.8.4 RECOMMENDED". This is not (semantically) what we _mean_.
What we mean is "use minimally 3.8.4 or ANY NEWER".

File: pom.xml (hunk 1 above)
Comment (cstamas) [reply pom.xml:97]: Same as above, 1.8.0-362 is NOT the recommended Java 8, just _minimal_ we allow to build with.

File: pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (cstamas) [reply pom.xml:96]: Hm, wait a second... seems am wrong: "[enforcer] redefined as a minimum version.". So did I misinterpret then?

Still, I think if we provide (semantically) equivalent value, but in form of (maven and not "redefined enforcer") range, we are somehow clearer what we _really mean_ here.

**Review by slawekjaranowski — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #1035

**Comment by jira-importer** [comment]
Resolve #1035

**Pull request merged**